src/models/train_model.py

```diff
@@ -1,61 +1,92 @@
-from src.data.AmazonReviewNew import AmazonReviewFullDataModule
-from src.models.bertsentimentclassifier import BertSentimentClassifier
-from transformers import BertTokenizer
+from src.data.AmazonReviewDataContinuousTokenization import AmazonReviewFullDataModule
+from src.models.distilbertsentimentclassifier import DistilBertSentimentClassifier
+from src.models.AzureMLLogger import AzureMLLogger
+from transformers import DistilBertTokenizer
 
 import pytorch_lightning as pl
 from pytorch_lightning.callbacks import ModelCheckpoint
 from pytorch_lightning.callbacks.early_stopping import EarlyStopping
-from pytorch_lightning.loggers import WandbLogger
 
-import torch
+import argparse
 
 
-def collate_fn(batch):
-    labels = torch.stack([x["labels"] for x in batch], axis=0)
-    input_ids = torch.stack([x["input_ids"] for x in batch], axis=0)
-    token_type_ids = torch.stack([x["token_type_ids"] for x in batch], axis=0)
-    attention_mask = torch.stack([x["attention_mask"] for x in batch], axis=0)
-    return {
-        "labels": labels,
-        "input_ids": input_ids,
-        "token_type_ids": token_type_ids,
-        "attention_mask": attention_mask,
+def main():
+    args = parse_args()
+    train_model(args)
+
+
+def parse_args():
+    parser = argparse.ArgumentParser(
+        description="Amazon review sentiment classification task"
+    )
+    parser.add_argument(
+        "--gpus",
+        type=int,
+        default=0,
+        metavar="N",
+        help="number of GPUs (default: 0)"
+    )
+    parser.add_argument(
+        "--epochs",
+        type=int,
+        default=3,
+        metavar="N",
+        help="number of epochs to train (default: 3)",
+    )
+    parser.add_argument(
+        "--batch-size",
+        type=int,
+        default=128,
+        metavar="N",
+        help="batch size (default: 128)",
+    )
+    parser.add_argument(
+        "--azure",
+        action="store_true",
+        default=False,
+        help="Run from Azure.",
+    )
+    args = parser.parse_args()
+    return args
+
+
+def train_model(args):
+    model_name = "distilbert-base-cased"
+    model = DistilBertSentimentClassifier(model_name)
+    tokenizer = DistilBertTokenizer.from_pretrained(model_name)
+
+    data = AmazonReviewFullDataModule(tokenizer, batch_size=args.batch_size)
+
+    trainer_params = {
+        "gpus": args.gpus,
+        "max_epochs": args.epochs,
+        "precision": 16 if args.gpus > 0 else 32,
```

Comment:
Using less memory is great, but why use 32 when we have 0 gpus? seems counter-intuitive

Reply:
CPU only allows for 32 and 64, it's in the documentation, and it gives an error if you use 16 on CPU